stats/opencensus/e2e_test.go

```diff
@@ -1338,6 +1338,53 @@ func (fe *fakeExporter) ExportSpan(sd *trace.SpanData) {
 	fe.seenSpans = append(fe.seenSpans, gotSI)
 }
 
+// waitForServerSpanAndSort waits until a server span appears somewhere in the
+// span list in an exporter, and then sorts the span list by putting the server
+// span first, the attempt span second, and overall call span third. Returns an
+// error if no server span found within 5 seconds, or if span list isn't length
+// 3 when server span is found.
+func waitForServerSpanAndSort(fe *fakeExporter) error {
```

Comment:
This seems extremely complicated and it very much depends on the test code in the one test case that needs it.

Why not just use `cmpopts.SortSlice` (sorted by `spanKind`) with the `cmp.Diff` calls below?

Reply:
Done. Reverified on forge with 10k runs. I tried using cmpopts, but I switched to sort.Slice instead, because I make assertions on the seenSpans past a pure cmp.Diff call (i.e. relationships between the three spans wrt child parent relationship checked by SpanID). This actually led me to discover a bug where my attempt span wasn't being emitted with Span Kind Client (sorting was failing, and that was why).